source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -62,6 +62,16 @@ FaultSettings::FaultSettings(const envoy::config::filter::http::fault::v2::HTTPF
     response_rate_limit_ =
         std::make_unique<Filters::Common::Fault::FaultRateLimitConfig>(fault.response_rate_limit());
   }
+
+  abort_percent_key_ = fault.abort_percent_key();
```

Comment:
This code can be significantly simplified by just using the existing default value if the key is not specified. I would probably add something like PROTOBUF_GET_STRING_OR_DEFAULT modeled after PROTOBUF_GET_MS_OR_DEFAULT.

Reply:
The default value is being constructed using the downstream cluster name. IIUC downstream cluster name is derived from the headers and is available in the `decodeHeaders` code path. Is it still doable in that case?

Reply:
I attempted doing this. Do take a look and LMK what you think